File: app/src/main/java/com/blackshift/verbis/adapters/HomePageBaseAdapter.java
package com.blackshift.verbis.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.blackshift.verbis.ui.fragments.WordListTitlesRecyclerFragment;

/**
 * Created by Devika on 13-03-2016.
 */
public class HomePageBaseAdapter extends FragmentStatePagerAdapter {

    public HomePageBaseAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position){
            //TODO: Use Xml.
            case 0: return "Wordlist";
            default: return "Wordlist";
        }
    }

    @Override
    public Fragment getItem(int position) {
        switch (position){
            case 0: return new WordListTitlesRecyclerFragment();
            default: return new WordListTitlesRecyclerFragment();
        }
    }

    @Override
    public int getCount() {
        return 1;
    }
}

File: app/src/main/java/com/blackshift/verbis/adapters/HomePageBaseAdapter.kt
package com.blackshift.verbis.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

import com.blackshift.verbis.ui.fragments.WordListTitlesRecyclerFragment
import com.blackshift.verbis.utils.annotations.PrivacyLevel

/**
 * Created by Devika on 13-03-2016.
 */
class HomePageBaseAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    override fun getPageTitle(position: Int): CharSequence {
        when (position) {
        //TODO: Use Xml.
            0 -> return "Wordlists"
            1 -> return "Your Wordlists"
            else -> return "Wordlist"
        }
    }

    override fun getItem(position: Int): Fragment {
        when (position) {
            0 -> return WordListTitlesRecyclerFragment.newInstance(PrivacyLevel.PUBLIC)
            1 -> return WordListTitlesRecyclerFragment.newInstance(PrivacyLevel.PRIVATE)
            else -> return WordListTitlesRecyclerFragment.newInstance(PrivacyLevel.PRIVATE)
        }
    }

    override fun getCount(): Int {
        return 2
    }
}
